--- DirectFB-extra-0.9.16/configure.in.orig	Mon Jan 13 16:19:59 2003
+++ DirectFB-extra-0.9.16/configure.in	Fri Jan 17 22:08:03 2003
@@ -101,7 +101,7 @@
 DFB_CFLAGS="$DFB_CFLAGS $DFB_INTERNAL_CFLAGS -DREENTRANT"
 
 if test "x$GCC" = xyes; then
-  DFB_CFLAGS="$DFB_CFLAGS -O3 -fexpensive-optimizations -ffast-math -Wall"
+  DFB_CFLAGS="$DFB_CFLAGS -ffast-math -Wall"
 fi
 
 
@@ -113,8 +113,6 @@
 if test "x$enable_debug" = xyes; then
     AC_DEFINE(DFB_DEBUG, 1, [Define to 1 to enable debugging support.])
     DFB_CFLAGS="$DFB_CFLAGS -g3"
-else
-    DFB_CFLAGS="$DFB_CFLAGS -g0"
 fi
 AM_CONDITIONAL(ENABLE_DEBUG, test "x$enable_debug" = xyes)
 
@@ -223,7 +229,10 @@
 # Honor aclocal flags
 ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
 
-
+if test "x$cflags_set" != "xset" ; then
+	DFB_CFLAGS="$DFB_CFLAGS $CFLAGS"
+fi
+    
 AC_SUBST(DFB_CFLAGS)
 AC_SUBST(DFB_LIBS)
 AC_SUBST(DATADIR)
--- DirectFB-extra-0.9.23/interfaces/IDirectFBImageProvider/Makefile.am.orig	Mon Oct 28 16:54:28 2002
+++ DirectFB-extra-0.9.23/interfaces/IDirectFBImageProvider/Makefile.am	Fri Jan 17 22:08:45 2003
@@ -3,7 +3,7 @@
 idirectfbimageproviderdir = $(MODULEDIR)/interfaces/IDirectFBImageProvider
 
 
-AM_CFLAGS = $(DFB_CFLAGS) $(SVG_CFLAGS) -DDATADIR=\"@DATADIR@\"
+AM_CFLAGS = $(DFB_CFLAGS) $(SVG_CFLAGS) -DDATADIR=\"@DATADIR@\" $(IMLIB2_INCLUDES)
 
 if IMLIB2
 IMLIB2_PROVIDER = libidirectfbimageprovider_imlib2.la
